ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+export PATH=$PATH:/tmp
+mkdir -p /tmp
+
+echo "$(date -u --rfc-3339=seconds) - Installing tools..."
+
+# Make sure jq is installed
+if ! command -v jq; then
+    # TODO move to image
+    curl -sL https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 > /tmp/jq
+    chmod +x /tmp/jq
+fi
+
+function cli_Login() {
+    # set the parameters we'll need as env vars
+    AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+    AZURE_AUTH_CLIENT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientId)"
+    AZURE_AUTH_CLIENT_SECRET="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientSecret)"
+    AZURE_AUTH_TENANT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .tenantId)"
+
+    # az should already be there
+    command -v az
+    az version
+
+    # log in with az
+    if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+        run_command "az cloud set --name AzureUSGovernment" || return 1
+    elif [[ "${CLUSTER_TYPE}" == "azure4" ]]; then
+        run_command "az cloud set --name AzureCloud" || return 1
+    elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+        echo "Unsupported cluster: ${CLUSTER_TYPE}"
+        exit 1
+    else
+        echo "Unexpected cluster: ${CLUSTER_TYPE}"
+        exit 1
+    fi
+    echo "$(date -u --rfc-3339=seconds) - Logging in to Azure..."
+    az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none || return 1
+}
+
+function getResouceGroup() {
+    local CLUSTER_NAME
+    CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+    run_command "az group list -o tsv | grep ${CLUSTER_NAME}"
+
+    if [[ $(az group list -o tsv | grep -c ${CLUSTER_NAME}) -gt 1 ]]; then
+        RESOURCE_GROUP=$(az group list -o tsv | grep ${CLUSTER_NAME} | grep -v "${CLUSTER_NAME}-rg" | awk '{print $4}')
+    else
+        RESOURCE_GROUP=$(az group list -o tsv | grep ${CLUSTER_NAME} | awk '{print $4}')
+    fi
+
+    run_command "az group show --name $RESOURCE_GROUP" || return 1
+    export RESOURCE_GROUP
+}
+
+function gatherLBs() {
+    local lbs aps output
+    run_command "az network lb list --resource-group $RESOURCE_GROUP > $OUTPUT_DIR/lbs.json"
+    lbs="$(az network lb list -g $RESOURCE_GROUP | jq -r '.[].name')"
+    for lb in $lbs; do
+        echo "loadbalance: $lb"
+        aps="$(az network lb address-pool list -g $RESOURCE_GROUP --lb-name ${lb} | jq -r '.[].name')"
+        for ap in $aps; do
+            echo "address-pool: $ap"
+            output="${OUTPUT_DIR}/address-${lb}-${ap}"
+            run_command "az network lb address-pool address list --lb-name $lb --pool-name $ap --resource-group $RESOURCE_GROUP -o table > $output"
+        done
+    done    
+}
+
+function gatherVMs() {
```

Comment:
I suggest to directly call the `run_command`, but not have to introduce a new function.